GPS.Data/Parser.cs

```diff
@@ -1,101 +1,160 @@
-﻿using System;
+﻿using GPS.Data.ParserObjects;
+using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Runtime.CompilerServices;
 using System.Text;
 using System.Xml.Serialization;
 
 namespace GPS.Data
 {
-    public class Parser
-    {
-        private readonly XmlSerializer Serializer;
+	public class Parser
+	{
+		private readonly XmlSerializer Serializer;
 
-        public GpsIsFile Outages { get; set; }
+		public GpsIsFile Outages { get; set; }
 
-        public Parser(string filePath)
-        {
-            //Serializes the .sof and populates each of the classes
-            Serializer = new XmlSerializer(typeof(GpsIsFile));
-            using Stream reader = new FileStream(filePath, FileMode.Open);
-            Outages = (GpsIsFile)Serializer.Deserialize(reader);
```

Comment:
Cannot rely on using statement to dispose if exception can be thrown from object other than the object in the using statement (eg, reader will only be disposed on an exception if reader itself throws an exception,  If Serializer.Deserialize can throw an exception reader.Dispose will not be called).  If this is the case, use try/finally patter with the object.Dispose called in the finally

Reply:
declare reader outside try as null
create it inside a try
null check
finally

Reply:
Don't use using anymore